OgraTests/OgraModel.swift

```diff
@@ -23,26 +25,30 @@ struct Pet {
 
 // MARK: - JSON Encoding and Decoding
 extension User: Decodable, Encodable {
-	static func create(id: Int)(name: String)(email: String?)(pet: Pet?) -> User {
-		return User(id:id, name:name, email:email, pet:pet)
-	}
-
-	static func decode(j: JSON) -> Decoded<User> {
-		return create
-			<^> j <|  "id"
-			<*> j <|  "name"
-			<*> j <|? "email"
-			<*> j <|? "pet"
-	}
-	
-	func encode() -> JSON {
-		return JSON.Object([
-			"id"    : self.id.encode(),
-			"name"  : self.name.encode(),
-			"email" : self.email.encode(),
-			"pet"   : self.pet.encode()
-		])
-	}
+    static func create(id: Int)(name: String)(email: String?)(pet: Pet?)(nicknames: [String]?)(accounts: [String:String]?) -> User {
+        return User(id:id, name:name, email:email, pet:pet, nicknames:nicknames, accounts:accounts)
+    }
+    
+    static func decode(j: JSON) -> Decoded<User> {
+        return create
+            <^> j <|   "id"
+            <*> j <|   "name"
+            <*> j <|?  "email"
+            <*> j <|?  "pet"
+            <*> j <||? "nicknames"
+            <*> .optional(flatReduce(["accounts"], initial: j, combine: decodedJSON) >>- Dictionary<String, String>.decode)
```

Comment:
> My approach here isn't yet a valid replacement for the previous (non-optional) dictionary extension.

This can be demonstrated by making `accounts` non-optional and updating the relevant type signatures as such.